pkg/extensions/tbtc/tbtc.go

```diff
@@ -19,15 +21,15 @@ var logger = log.Logger("tbtc-extension")
 const maxActAttempts = 3
 
 // Initialize initializes extension specific to the TBTC application.
-func Initialize(ctx context.Context, handle Handle) error {
+func Initialize(ctx context.Context, chain Handle) error {
 	logger.Infof("initializing tbtc extension")
 
-	tbtc := &tbtc{handle}
+	tbtc := newTBTC(chain)
 
 	err := tbtc.monitorRetrievePubKey(
 		ctx,
 		exponentialBackoff,
-		150*time.Minute,
+		150*time.Minute, // 30 minutes before the 3 hours on-chain timeout
```

Comment:
Why not 15 minutes before just like for `monitorProvideRedemptionSignature`? `maxActionAttemps = 3` and we have backoff of 2, 4, and 8 seconds. We also have the transaction monitoring process increasing gas price for transaction in the pool.